internal/providers/terraform/azure/app_service_environment.go

```diff
@@ -0,0 +1,101 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+
+	"github.com/shopspring/decimal"
+)
+
+func GetAzureRMAppIsolatedServicePlanRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_app_service_environment",
+		RFunc: NewAzureRMAppIsolatedServicePlan,
+		ReferenceAttributes: []string{
+			"resource_group_name",
+		},
+		Notes: []string{
+			"Costs associated with running an app service plan in Azure",
+		},
+	}
+}
+
+func NewAzureRMAppIsolatedServicePlan(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: AppIsolatedServicePlanCostComponent(d, u),
+	}
+}
+func AppIsolatedServicePlanCostComponent(d *schema.ResourceData, u *schema.UsageData) []*schema.CostComponent {
+
+	tier := d.Get("pricing_tier").String()
+	group := d.References("resource_group_name")
+	location := group[0].Get("location").String()
+	stampFeeTiers := []string{"I1", "I2", "I3"}
+	productName := "Isolated Plan"
+	purchaseOption := "Consumption"
+	costComponents := make([]*schema.CostComponent, 0)
+	os := "Linux"
+	if u != nil && u.Get("operating_system").Exists() {
```

Comment:
```suggestion
	if u != nil && u.Get("operating_system").Type != gjson.Null {
```

Search the contributing guide for `gjson.Null` to see why we do it this way.